src/Plugins/BotSharp.Plugin.LangChain/Providers/ChatCompletionProvider.cs

```diff
@@ -0,0 +1,72 @@
+using BotSharp.Abstraction.Agents.Enums;
+using BotSharp.Abstraction.Agents.Models;
+using BotSharp.Abstraction.Conversations.Models;
+using BotSharp.Abstraction.Loggers;
+using BotSharp.Abstraction.MLTasks;
+using LangChain.Providers;
+using LangChain.Providers.Google.VertexAI;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace BotSharp.Plugin.VertexAI.Providers
+{
+    public class ChatCompletionProvider(
+    VertexAIConfiguration config,
+    ChatSettings settings,
+    ILogger<TextCompletionProvider> logger,
+    IServiceProvider services) : IChatCompletion
+    {
+        public string Provider => "vertexai";
+        private readonly VertexAIConfiguration _config = config;
+        private readonly ChatSettings? _settings = settings;
+        private readonly IServiceProvider _services = services;
+        private readonly ILogger _logger = logger;
+        public required string _model;
+
+        public void SetModelName(string model)
+        {
+            _model = model;
+        }
+
+        public async Task<RoleDialogModel> GetChatCompletions(Agent agent, List<RoleDialogModel> conversations)
+        {
+            var hooks = _services.GetServices<IContentGeneratingHook>().ToList();
```

Comment:
We can use `HookEmitter.Emit` to simplify the Hook invocation

Reply:
Is there any example of use? I didn't find it in the rest of the implementations (HookEmitter.Emit)

Regarding the tools, they are configured through VertexAIConfiguration (Test: ToolChat). Would it be worth doing it like this?
https://github.com/tryAGI/LangChain.Providers/blob/main/src/Google.VertexAI/test/VertexAITest.cs